awx/ui_next/src/screens/Inventory/InventoryGroups/InventoryGroup/InventoryGroup.jsx

```diff
@@ -0,0 +1,142 @@
+  const [hasContentLoading, setHasContentLoading] = useState(true);
+  const [contentError, setHasContentError] = useState(null);
+
+  useEffect(() => {
+    const loadData = async () => {
+      try {
+        const { data } = await GroupsAPI.readDetail(match.params.groupId);
+        setInventoryGroup(data);
+        setBreadcrumb(inventory, data);
+      } catch (err) {
+        setHasContentError(err);
+      } finally {
+        setHasContentLoading(false);
+      }
+    };
+
+    loadData();
+  }, [
+    history.location.pathname,
+    match.params.groupId,
+    inventory,
+    setBreadcrumb,
+  ]);
+
+  const tabsArray = [
+    {
+      name: i18n._(t`Return to Groups`),
+      link: `/inventories/inventory/${inventory.id}/groups`,
+      id: 99,
+      isNestedTabs: true,
+    },
+    {
+      name: i18n._(t`Details`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/details`,
+      id: 0,
+    },
+    {
+      name: i18n._(t`Related Groups`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/nested_groups`,
+      id: 1,
+    },
+    {
+      name: i18n._(t`Hosts`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/nested_hosts`,
+      id: 2,
+    },
+  ];
+  if (contentError) {
+    return <ContentError error={contentError} />;
+  }
+  if (hasContentLoading) {
+    return <ContentLoading />;
+  }
+
+  let cardHeader = null;
+  if (
+    history.location.pathname.includes('groups/') &&
+    !history.location.pathname.endsWith('edit')
+  ) {
+    cardHeader = (
+      <CardHeader style={{ padding: 0 }}>
+        <RoutedTabs history={history} tabsArray={tabsArray} />
+        <CardCloseButton
+          linkTo={`/inventories/inventory/${inventory.id}/group`}
+        />
+      </CardHeader>
+    );
+  }
+
+  return (
+    <>
+      {cardHeader}
+      <Switch>
+        <Redirect
+          from="/inventories/inventory/:id/groups/:groupId"
+          to="/inventories/inventory/:id/groups/:groupId/details"
+          exact
+        />
+        {inventoryGroup && [
+          <Route
+            key="edit"
+            path="/inventories/inventory/:id/groups/:groupId/edit"
+            render={() => {
+              return (
+                <InventoryGroupEdit
+                  inventory={inventory}
+                  inventoryGroup={inventoryGroup}
+                />
+              );
+            }}
+          />,
+          <Route
+            key="details"
+            path="/inventories/inventory/:id/groups/:groupId/details"
+            render={() => {
+              return <InventoryGroupDetail inventoryGroup={inventoryGroup} />;
+            }}
+          />,
+        ]}
+        <Route
+          key="not-found"
+          path="*"
+          render={() => {
+            return (
+              !hasContentLoading && (
```

Comment:
We check `hasContentLoading` above and return early if it is true so it doesn't need to be checked again.